internal/entities/quote/pegin_quote.go

```diff
@@ -30,14 +29,22 @@ type PeginQuoteRepository interface {
 	InsertQuote(ctx context.Context, quote CreatedPeginQuote) error
 	GetQuote(ctx context.Context, hash string) (*PeginQuote, error)
 	GetPeginCreationData(ctx context.Context, hash string) PeginCreationData
+	GetQuotes(ctx context.Context, hashes []string) ([]PeginQuote, error)
 	GetRetainedQuote(ctx context.Context, hash string) (*RetainedPeginQuote, error)
 	InsertRetainedQuote(ctx context.Context, quote RetainedPeginQuote) error
 	UpdateRetainedQuote(ctx context.Context, quote RetainedPeginQuote) error
 	GetRetainedQuoteByState(ctx context.Context, states ...PeginState) ([]RetainedPeginQuote, error)
+	GetQuotesByState(ctx context.Context, filter GetPeginQuotesByStateFilter) ([]PeginQuote, error)
```

Comment:
this definition is not accurate because its called `GetQuotesByState` and you're using the date + the state to filter. It is not clear for the consumer, IMO you have two options, either rename it to reflect the proper functionality of the function, or to go with a better solution like the functional options pattern to make "complex" queries